.devcontainer/devcontainer.json

```diff
@@ -1,17 +1,36 @@
 // If you want to run as a non-root user in the container, see .devcontainer/docker-compose.yml.
 {
 	"name": "Go",
-	"dockerComposeFile": "docker-compose.yml",
-    "service": "docker-in-docker",
+	"dockerFile": "./Dockerfile",
 	"workspaceFolder": "/workspace",
+	"context": "..",
+	"workspaceMount": "src=${localWorkspaceFolder},dst=/workspace,type=bind",
 	"postCreateCommand": "make set-kindcluster",
-    "shutdownAction": "stopCompose",
+    "runArgs": [
+		// Mount the env file
+		"--env-file", "${localWorkspaceFolder}/.devcontainer/.env",
+		// Mount go mod cache
+		"-v", "dboperator-gomodcache:/go/pkg",
+		// Cache vscode exentsions installs and homedir
+		"-v", "dboperator-vscodecache:/root/.vscode-server",
```

Comment:
The problem with caching the extensions is on consecutive rebuilds, `postCreateCommand` will not fire off as detailed [here](https://code.visualstudio.com/docs/remote/containers-advanced#_avoiding-extension-reinstalls-on-container-rebuild). Unless the kind config is saved between rebuilds, you'll need to run `make set-kindlcuster` every time which if you're going to need to do that then might as well remove `postCreateCommand` and document somewhere saying you'll need to run the command yourself.

The other thing with caching the extensions, newly added extensions will not be installed after the cache has been created. This could be a bit of an annoyance for developers if they're not aware of the reason and what needs to be done to fix it. Would you be able to document that somewhere? Either as a comment or in one of the README files would do.

Reply:
Thanks for taking a look Jason. Yup these are the correct trade-offs with the cached extensions, given we're working on shared wifi with customers it felt like a reasonable price to pay as it allowed a mostly offline experience and near instant start of the devcontainer (as long as you've run it once before somewhere) vs a long wait for extension download. 

I'll add a section to the docs around how to reset the cached extensions install. 

In terms of the `postCreateCommand` I'll remove this from the `.devcontainer` as the docs already direct the user to run `make set-kindcluster` after the devcontainer has started so I don't think there is any need to update them. 

```markdown
6. Once the container has finished building, open a new terminal in VS Code (``Control+Shift+` ``) and type:
	```shell
	make set-kindcluster
	```
```
https://github.com/lawrencegripper/azure-databricks-operator/blob/21398ff/docs/contributing.md#L48

Sound like a good plan?

Reply:
Changes pushed, hopefully good to go, any feedback let me know.